From bd028a84191e2cc3f21ccdb8a5ddbdab743b81e6 Mon Sep 17 00:00:00 2001
From: rofl0r <retnyg@gmx.net>
Date: Wed, 2 Nov 2016 09:30:38 +0000
Subject: [PATCH] link_open: do not crash when run as non-root

if there's a permission error (or another error), the returned handle
will be null and the subsequent dereference will crash the program.
---
 link-packet-socket.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/link-packet-socket.c b/link-packet-socket.c
index dac1bbb..bc22fbe 100644
--- a/link-packet-socket.c
+++ b/link-packet-socket.c
@@ -118,6 +118,10 @@ get_hardware_address(const char *if_name, unsigned char hw_address[]) {
    link_t *handle;
 
    handle = link_open(if_name);
+   if(!handle) {
+       err_sys("link_open");
+       return;
+   }
 
 /* Obtain hardware address for specified interface */
    if ((ioctl(handle->fd, SIOCGIFHWADDR, &(handle->ifr))) != 0)
